aws/resource_aws_codebuild_webhook_test.go

```diff
@@ -0,0 +1,78 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/codebuild"
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAwsCodeBuildWebhook_basic(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsCodeBuildWebhookDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCodeBuildWebhookConfig_basic(acctest.RandString(5)),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsCodeBuildWebhookExists("aws_codebuild_webhook.test"),
+					resource.TestCheckResourceAttrSet("aws_codebuild_webhook.test", "url"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAwsCodeBuildWebhookDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).codebuildconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_codebuild_webhook" {
+			continue
+		}
+
+		resp, err := conn.BatchGetProjects(&codebuild.BatchGetProjectsInput{
+			Names: []*string{
+				aws.String(rs.Primary.ID),
+			},
+		})
+
+		if err != nil {
+			return err
```

Comment:
A resource not found exception here would be a good thing and should `return nil` -- will fix on merge.